docs-web/configuration/ingress-resources/advanced-configuration-with-snippets.md

```diff
@@ -0,0 +1,107 @@
+# Advanced Configuration with Snippets
+
+Snippets allow you to insert raw NGINX config into different contexts of the NGINX configurations that the Ingress Controller generates. These should be used a last-resort solution in cases where annotations and ConfigMap entries cannot help. Snippets are intended for advanced NGINX users who need more control over the generated NGINX configuration.
+
+Snippets are also available through the [ConfigMap](/nginx-ingress-controller/configuration/global-configuration/configmap-resource). Annotations take precedence over the ConfigMap.
+
+## Using Snippets
+
+The example below shows how to use snippets to customize the NGINX configuration template using annotations.
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: cafe-ingress-with-snippets
+  annotations:
+    nginx.org/server-snippets: |
+      location / {
+          return 302 /coffee;
+      }
+    nginx.org/location-snippets: |
+      add_header my-test-header test-value;
+spec:
+  rules:
+  - host: cafe.example.com
+    http:
+      paths:
+      - path: /tea
+        backend:
+          serviceName: tea-svc
+          servicePort: 80
+      - path: /coffee
+        backend:
+          serviceName: coffee-svc
+          servicePort: 80
+```
+
+Generated NGINX configuration:
+```nginx
+server {
+    listen 80;
+
+
+    location / {
+        return 302 /coffee;
+    }
+
+
+    location /coffee {
+        proxy_http_version 1.1;
+
+
+        add_header my-test-header test-value;
+        ...
+        proxy_pass http://default-cafe-ingress-with-snippets-cafe.example.com-coffee-svc-80;
+    }
+
+    location /tea {
+        proxy_http_version 1.1;
+        
+        add_header my-test-header test-value;
+        ...
+        proxy_pass http://default-cafe-ingress-with-snippets-cafe.example.com-tea-svc-80;
+    }
+}
+```
+**Note**: The generated configs are truncated for the clarity of the example.
+
+## Summary of Snippets
+
+See the [snippets annotations](/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-annotations/#snippets-and-custom-templates) documentation for more information.
+
+## Disadvantages of using snippets
+
```

Comment:
could you add a sentence before the list?

Snippets have the following disadvantages:
* *Complexity* ...